framework/include/utils/MooseTypes.h

```diff
@@ -166,6 +166,8 @@ enum KernelType
 {
   KT_TIME = 0,
   KT_NONTIME = 1,
+  KT_NONEIGEN = 2,
+  KT_EIGEN  = 3,
```

Comment:
I appreciate that in general. However sometimes we run into bad designs as we add new capability. I think it's worth refactoring those designs in additional commits as we go. It helps make things more maintainable and I'm all for doing that.